kafka-ui-api/src/main/java/com/provectus/kafka/ui/serde/schemaregistry/FallbackMessageFormatter.java

```diff
@@ -0,0 +1,47 @@
+package com.provectus.kafka.ui.serde.schemaregistry;
+
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.dataformat.smile.databind.SmileMapper;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class FallbackMessageFormatter implements MessageFormatter {
+  private static final SmileMapper smileMapper = new SmileMapper();
+  private static final JsonMapper jsonMapper = new JsonMapper();
+
+  @Override
+  public String format(String topic, byte[] value) {
+    if (value == null) {
+      return null;
+    }
+    String decodedString = new String(value, StandardCharsets.UTF_8);
+    if (isJsonLike(decodedString)) {
+      try {
+        return jsonMapper.readTree(value).toPrettyString();
+      } catch (IOException e) {
+        return decodedString;
+      }
```

Comment:
I think it is redundant to try to parse as json and prettify it - it will be done in ui again. Let's remove json parsing logic and leave smile checks only

Reply:
ok, agree with Roman, that it better to implement it as a pluggable serde

Reply:
Actually, I only added the JSON formatting stuff to make it appeal to a wider audience, like I thought "maybe smile format is something considered esoteric and hence does not belong straight in the code". But it was the simplest way to achieve it since I do not use any schema registry or anything else to signify that the messages for this topic is in smile format. So then the autodetection part came along.

Anyhow, this MR can be closed if there is a plugin architecture so I can customize it that way. In the meantime we are just running a fork.

Reply:
I'll keep the PR open so I won't forget about it once we implement serde API. I'll let you know once we do that if you're willing to adapt it in the future.

Reply:
👍